src/ucs/vfs/fuse/vfs_fuse.c

```diff
@@ -164,13 +164,36 @@ static int ucs_vfs_fuse_release(const char *path, struct fuse_file_info *fi)
     return 0;
 }
 
+static int ucs_vfs_fuse_write(const char *path, const char *buf, size_t size,
+                              off_t offset, struct fuse_file_info *fi)
+{
+    char *data = (void*)fi->fh;
+    ucs_string_buffer_t strb;
+    ucs_status_t status;
+
+    ucs_string_buffer_init(&strb);
+    ucs_string_buffer_appendf(&strb, "%.*s", (int)offset, data);
+    ucs_string_buffer_appendf(&strb, "%s", buf);
+
+    status = ucs_vfs_path_write_file(path, &strb);
```

Comment:
if the file is written in multiple chunks does it mean a callback is invoked every time?

Reply:
Yes, it does.
It's not an issue for current files, because of a small size.

We can update file content pointed by `fi->fh` and save the changes in `release` method.
Or, we can pass `const char *buf, size_t size, off_t offset` further to `ucs_vfs_path_write_file`.

The first one seems to be consistent from pairing open/release and read/write methods. However, it requires to distinguish read/write modes using open flags in release method.

Reply:
IMO, release() is a problem: we need to be able to return an error if the value is invalid (for example writing a string to a numeric variable). See what happens with procfs/sysfs for example. I think we should write to the file after encountering "\n" in the text.
This also means the write callbacks should return ucs_status_t and then it should be propagated to the return value of fuse write callback

Reply:
Added return value to VFS node's write callback, and VFS obj write interface.
Implemented writing the data only incase of offset is 0 to simplify the logic and keep interfaces more clear.

Reply:
We add '\n' in VFS node read callback. I will add common write callbacks to properly deal with '\n' in the next PR.